elixir/lib/media_player/commands.ex

```diff
@@ -0,0 +1,121 @@
+defmodule MediaPlayer.Commands do
+  use Task
+
+  alias MediaPlayer.Actions, as: Actions
+  alias MediaPlayer.Config.Connection, as: Connection
+
+  defp keyspace, do: Connection.keyspace()
+  defp table, do: Connection.table()
+
+  def add_from(title, album, artist, created) do
+    query =
+      "INSERT INTO #{keyspace()}.#{table()} (id, title, album, artist, created_at) VALUES (?, ?, ?, ?, ?);"
+
+    {:ok, created, _} = DateTime.from_iso8601(created <> "T00:00:00Z")
+
+    Actions.run_query(query, [UUID.uuid4(), title, album, artist, created])
+
+    IO.puts("Song added!")
+  end
+
+  def add() do
+    title = IO.gets("Enter the title of the song: ") |> String.trim()
+    album = IO.gets("Enter the album of the song: ") |> String.trim()
+    artist = IO.gets("Enter the artist of the song: ") |> String.trim()
+
+    created =
+      IO.gets("Enter the date the song was created (YYYY-MM-DD): ")
+      |> String.trim()
+
+    add_from(title, album, artist, created)
+  end
+
+  def list do
+    query = "SELECT id, title, album, artist, created_at FROM #{keyspace()}.#{table()};"
+
+    Actions.run_query(query)
+    |> Enum.each(fn %{
+                      "id" => id,
+                      "title" => title,
+                      "album" => album,
+                      "artist" => artist,
+                      "created_at" => created_at
+                    } ->
+      IO.puts(
+        "ID: #{id} | Title: #{title} | Album: #{album} | Artist: #{artist} | Created At: #{created_at}"
+      )
+    end)
+  end
+
+  def delete_from(index) do
+    query = "SELECT id, created_at FROM #{keyspace()}.#{table()};"
```

Comment:
This query doesn't make sense since we're talking about ScyllaDB.

I mean, theoretically in a small system like these, you can query and will receive the same result from a "SELECT * FROM table". But dealing with Scylla, that has a huge throughput and the list will not be same never (counting as index).

Maybe if you store the items in a list, fetch the id and reuse it after it will make more sense...

PS: I'm not familiar with Elixir, but I think that sounds reasonable.

Reply:
Yes, that makes sense. I believe I hadn't yet checked all the possibilities and ended up not being aware of this detail.

Resolved! Check this https://github.com/scylladb/scylla-cloud-getting-started/pull/22/commits/8587dd909ae9345b74fe0620153508d6ef088fb7